econml/two_stage_least_squares.py

```diff
@@ -250,7 +330,7 @@ def marginal_effect(self, T, X=None):
         ft_X = self._x_featurizer.fit_transform(X)
         n = shape(T)[0]
         dT = self._dt_featurizer.fit_transform(T)
-        W = np.zeros((n, self._d_w))
+        W = np.zeros((size(T), self._d_w))
```

Comment:
This is the only real change in this PR - the shape W was wrong when there was more than a single treatment.

The other changes are shoring up our tests for this class and adding a `DPolynomialFeatures` class that can be used for `dt_features` when `t_features` is a `PolynomialFeatures` instance.